monitoring.go

```diff
@@ -123,6 +134,19 @@ os hostname <i>{{.Hostname}}</i><br>
 </html>
 `))
 
+func exitHandler(srv *http.Server) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != "POST" {
+			http.Error(w, "Use POST method for exiting", http.StatusMethodNotAllowed)
+			return
+		}
+
+		log.Infof("Received /exit")
+		http.Error(w, "OK exiting", http.StatusOK)
+		go srv.Shutdown(context.Background())
```

Comment:
http.Error was in fact not synchronous, so I had to use `http.Server.Shutdown` which is a graceful shutdown. This makes the code a bit more complex but there should be no race.